backend/app/api/v1/endpoints/cpt/cptJobs.py

```diff
@@ -25,161 +26,191 @@
     "ocm": ocmMapper,
 }
 
+productsFilter = {
+    "ocp": ocpFilter,
+    "quay": quayFilter,
+    "hce": hceFilter,
+    "telco": telcoFilter,
+    "ocm": ocmFilter,
+}
+
+
+### **Helper Function for Default Date Handling**
+def get_default_dates(start_date, end_date):
+    today = datetime.utcnow().date()
+    return (start_date or today - timedelta(days=5), end_date or today)
+
+
+async def fetch_data_limited(product, *args, **kwargs):
+    return await fetch_data(product, *args, **kwargs)
+
+
+async def fetch_data(
+    product, start_date, end_date, size=None, offset=None, filter=None, is_filter=False
+):
+    try:
+        fetch_function = productsFilter[product] if is_filter else products[product]
+        args = (
+            (start_date, end_date, filter)
+            if is_filter
+            else (start_date, end_date, size, offset, filter)
+        )
+
+        response = await fetch_function(*args)
+
+        if not response:
+            return {"data": pd.DataFrame(), "total": 0} if not is_filter else {}
+
+        return {
+            "data": (
+                response["data"] if not is_filter else response.get("filterData", [])
+            ),
+            "total": response["total"],
+            "summary": response.get("summary", {}),
+        }
+    except Exception as e:
+        print(f"Error fetching data for {product}: {e}\n{traceback.format_exc()}")
```

Comment:
If this is really a "crazy unexpected error" I don't like just hiding it in a backend printout nobody's likely to see -- and which won't have any useful context if someone does. If this is really an unexpected "internal service error", then I think we should just let the FastAPI fallback handle it. We'll get an HTTP 500 response and the server log will show the traceback to help diagnosis.